warehouse/query-core/src/main/java/datawave/query/util/GeoWaveUtils.java

```diff
@@ -0,0 +1,860 @@
+     * 
+     * @param startGeohash
+     * @param endGeohash
+     * @param longBuffer
+     *            a reusable byte buffer of Long.BYTES length
+     * @return
+     */
+    public static Geometry rangeToGeometry(String startGeohash, String endGeohash, ByteBuffer longBuffer) {
+        return rangeToGeometry(decodeTier(startGeohash), decodePosition(startGeohash), decodePosition(endGeohash), longBuffer);
+    }
+    
+    /**
+     * Given a range defined by byteArrayRange, this will generate a Geometry which represents that range.
+     * 
+     * @param byteArrayRange
+     * @param longBuffer
+     *            a reusable byte buffer of Long.BYTES length
+     * @return
+     */
+    public static Geometry rangeToGeometry(ByteArrayRange byteArrayRange, ByteBuffer longBuffer) {
+        return rangeToGeometry(decodeTier(byteArrayRange.getStart()), decodePosition(byteArrayRange.getStart()), decodePosition(byteArrayRange.getEnd()),
+                        longBuffer);
+    }
+    
+    /**
+     * Given a range at a given tier, this will generate a Geometry which represents that range.
+     * 
+     * @param tier
+     * @param start
+     * @param end
+     * @param longBuffer
+     *            a reusable byte buffer of Long.BYTES length
+     * @return
+     */
+    public static Geometry rangeToGeometry(int tier, long start, long end, ByteBuffer longBuffer) {
+        longBuffer = initLongBuffer(longBuffer);
+        
+        GeometryFactory gf = new GeometryFactory();
+        
+        List<ByteArrayId> byteArrayIds = decomposeRange(tier, start, end, longBuffer);
+        
+        List<Geometry> geometries = new ArrayList<>(byteArrayIds.size());
+        for (ByteArrayId byteArrayId : byteArrayIds) {
+            MultiDimensionalNumericData bounds = GeometryNormalizer.indexStrategy.getRangeForId(byteArrayId);
+            
+            if (decodeTier(byteArrayId) <= 1) {
+                // @formatter:off
+                geometries.add(gf.toGeometry(
+                        new Envelope(
+                                bounds.getMinValuesPerDimension()[0],
+                                bounds.getMaxValuesPerDimension()[0],
+                                Math.max(-90, bounds.getMinValuesPerDimension()[1]),
+                                Math.min(90, bounds.getMaxValuesPerDimension()[1]))));
+                // @formatter:on
+            } else if (inBounds(bounds)) {
+                // @formatter:off
+                geometries.add(gf.toGeometry(
+                        new Envelope(
+                                bounds.getMinValuesPerDimension()[0],
+                                bounds.getMaxValuesPerDimension()[0],
+                                bounds.getMinValuesPerDimension()[1],
+                                bounds.getMaxValuesPerDimension()[1])));
+                // @formatter:on
+            }
+        }
```

Comment:
Wondering if all the instances of this pattern could be consolidated into one or two helper methods, similar to your `positionToGeometry` function

Reply:
I think the latest commit has this covered.